packages/vm/src/imports.rs

```diff
@@ -447,7 +447,7 @@ pub fn do_debug<A: BackendApi + 'static, S: Storage + 'static, Q: Querier + 'sta
         let message_data = read_region(&data.memory(&mut store), message_ptr, MAX_LENGTH_DEBUG)?;
         let msg = String::from_utf8_lossy(&message_data);
         let gas_remaining = data.get_gas_left(&mut store);
-        (*debug_handler)(
+        debug_handler.borrow_mut()(
```

Comment:
Can the same be achieved with https://doc.rust-lang.org/std/rc/struct.Rc.html#method.get_mut directly? The `Rc<RefCell<` feels redundant. Not sure, but would one of the two work?

```
    # either
    debug_handler: Option<Rc<DebugHandlerFn>>,
    
    # or
    debug_handler: Option<RefCell<DebugHandlerFn>>,
```

But I might be wrong about this

Reply:
After double-checking, I believe we need some kind of pointer around the `RefCell`, because otherwise the type is unsized (internally it contains a `Cell`, which directly contains the `DebugHandlerFn`, which has no fixed size).
`RefCell` also cannot be cloned (at least without cloning the contained thing).

We also cannot use the `Rc::get_mut` function, because that returns `None` if we have multiple `Rc`s of the debug handler. We always keep one `Rc` in the context data and clone it in order to use it in this line, so we always have multiple `Rc`s here. Since this would allow holding an immutable and mutable borrow at the same time.

[Rc<RefCell<_>> is apparently also a common pattern](https://doc.rust-lang.org/std/cell/index.html#introducing-mutability-inside-of-something-immutable)